### PR TITLE
[Core] Create Modelers

### PR DESCRIPTION
**Pull request by tteschemacher**
Dear all,

this PR allows to Create Modelers in Python with the CreateModeler function.

Thx for the review!

### CODE REVIEW

**Comment by tteschemacher** [comment]
@KratosMultiphysics/technical-committee this is also ready to be merged.

File: kratos/modeler/modeler.h [hunk 1]
```diff
@@ -16,45 +16,26 @@
...
-
-///@}
 ///@name Kratos Classes
 ///@{
 
-/// Short class definition.
-/** Detail class definition.
+/// Modeler to interact with ModelParts.
+/* The modeler is designed to interact, create and update
+   the ModelPart of the analyses after and at certain steps.
```
Comment (philbucher) [inline kratos/modeler/modeler.h]:
```suggestion
/* The modeler is designed to create the ModelParts of the analyses
```

File: kratos/modeler/modeler.h [hunk 2]
```diff
@@ -194,107 +135,24 @@ class Modeler
...
-    ///@name Un accessible methods
-    ///@{
-
-    /// Assignment operator.
-    Modeler& operator=(Modeler const& rOther);
-
-    /// Copy constructor.
-    Modeler(Modeler const& rOther);
-
+    SizeType mEchoLevel;
```
Comment (philbucher) [inline kratos/modeler/modeler.h]:
```suggestion
    SizeType mEchoLevel = 0;
```

File: kratos/modeler/modeler_factory.h [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+#include "includes/kratos_components.h"
+#include "modeler/modeler.h"
+
+namespace Kratos
+{
+///@name Kratos Classes
+///@{
+
+/// Creates Modelers.
+namespace ModelerFactory
```
Comment (philbucher) [inline kratos/modeler/modeler_factory.h:30]: this should be merged with #3185 (separate PR)

**Review by philbucher — COMMENTED** [review]
minor comments then I will approve

**Comment by philbucher** [comment]
I am confused now, what is the difference btw this PR and #6605. Seems like both have the same changes

Can you please close the PRs that are no longer relevant

File: kratos/modeler/modeler_factory.h (hunk 3 above)
Comment (tteschemacher) [reply kratos/modeler/modeler_factory.h:30]: Ok but we can do this as soon as this PR is merged?

**Review by philbucher — APPROVED** [review]
Nice work :+1:
approving to not block further developments
HOWEVER
this will need to be updated/integrated in #3185

**Comment by tteschemacher** [comment]
Sweet. Thanks a lot!

**Pull request merged**